pkg/apis/servicecatalog/types.go

```diff
@@ -144,11 +144,16 @@ type CommonServiceBrokerSpec struct {
 //
 // ServiceClass allowed property names:
 //   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServiceClass.Spec.ExternalID
+//   spec.serviceBrokerName - the value set to [Cluster]ServiceClass.Spec.[Cluster]ServiceBrokerName
 //
 // ServicePlan allowed property names:
 //   name - the value set to [Cluster]ServiceClass.Name
-//   externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalName - the value set to [Cluster]ServiceClass.Spec.ExternalName
+//   spec.externalID - the value set to [Cluster]ServiceClass.Spec.ExternalID
+//   spec.serviceClassName - the value set to [Cluster]ServiceClass.Spec.ServiceClassRef.Name
```

Comment:
These comments are incorrect, this is the ServicePlan object so the properties map from [Cluster]ServicePlan.Spec.ExternalName

Reply:
`spec.serviceClassName` was not intended to be exposed as an option, it does not make sense to filter a plan based on service class. It is more for knowing which plans to remove when a class is removed.

Reply:
I am trying to decide if filtering by service broker name on plans should be allowed... I think they should be.

Reply:
> filtering by service broker name on plans should be allowed.

Looks like the code implemented this, just not documented.